rlp/encode.go

```diff
@@ -428,9 +429,13 @@ func writeBigIntNoPtr(val reflect.Value, w *encbuf) error {
 	return writeBigInt(&i, w)
 }
 
+var (
+	errRLPEncodeWantPositiveBigInt = errors.New("rlp: cannot encode negative *big.Int")
```

Comment:
technically should be `WantNonNegative`. Maybe `errRLPEncodeNegativeBigInt` is sufficient